_internal/utils/cache.ts

```diff
@@ -63,8 +63,10 @@ export const initCache = <Data = any>(
     }
     const setter = (key: string, value: any, prev: any) => {
       provider.set(key, value)
-      const subs = subscriptions[key]
-      if (subs) {
+      if (subscriptions[key]) {
+        // Make a copy of subscriptions to avoid mutations of subscriptions to affect
+        // the array while looping
+        const subs = subscriptions[key].slice()
```

Comment:
```suggestion
const subs = subscriptions[key]
      if (subs) {
        for (const fn of subs) {
          fn(value, prev)
        }
      }
```
Maybe we could use `for of` here, so it don't need a extra copy of subs here ?

Reply:
Yeah, that should probably achieve the same. I can update on Monday as soon as I have access to a computer! Thanks for feedback and attention!